File: api/utils/overall.py


def calculate_team_overall(positions, players, modifiers):

    total = 0

    player_ids = [str(p["player"]) for p in positions]

    for player_id in player_ids:
        total += players[player_id][1]

    for modifier in modifiers:
        if modifier["target"]["type"] == "players" and "ids" in modifier["target"]:
            for m_id in modifier["target"]["ids"]:
                if str(m_id) in player_ids:
                    for value in modifier["values"]:
                        if value["field"] in ["ovr", "overall"]:
                            total += value["value"]

    return total

File: web-coach/src/utils/overall.js
export function calculateTeamOverall(positions, players, modifiers) {
  let total = 0;

  // Get player IDs from positions array
  const playerIds = positions.map((p) => String(p.player));

  // Sum base player values (assuming players is an object with playerId keys and [something, value] as value)
  playerIds.forEach((playerId) => {
    if (players[playerId]) {
      total += players[playerId][1];
    }
  });

  // Add modifiers if they target players in this lineup
  modifiers.forEach((modifier) => {
    const target = modifier.target;
    if (target.type === "players" && Array.isArray(target.ids)) {
      target.ids.forEach((modId) => {
        if (playerIds.includes(String(modId))) {
          modifier.values.forEach((value) => {
            if (value.field === "ovr" || value.field === "overall") {
              total += value.value;
            }
          });
        }
      });
    }
  });

  return total;
}
